ctl/src/system_info.rs

```diff
@@ -0,0 +1,68 @@
+use anyhow::{anyhow, Context, Result};
+use serde_derive::Deserialize;
+use std::env;
+
+#[derive(Deserialize)]
+struct SystemInfo {
+    git_sha: String,
+    git_commit_date: String,
+}
+
+fn read_system_info() -> Result<SystemInfo> {
+    let content = std::fs::read_to_string("/etc/system-info.toml")
+        .context("fail to read /etc/system-info.toml")?;
+    Ok(toml::from_str::<SystemInfo>(&content)?)
+}
+
+fn bitcoind_version() -> Result<String> {
+    let output = std::process::Command::new("bitcoind")
+        .args(["--version"])
+        .output()
+        .context("could not run bitcoind command")?;
+    std::str::from_utf8(&output.stdout)?
+        .split('\n')
+        .next()
+        .and_then(|line| line.split("Bitcoin Core version ").nth(1))
+        .ok_or(anyhow!(
+            "failed to parse version from bitcoind command output"
+        ))
+        .map(|version| version.into())
+}
+
+fn cockroach_version() -> Result<String> {
+    let output = std::process::Command::new("cockroach")
+        .args(["version"])
+        .output()
+        .context("could not run cockroach command")?;
+    std::str::from_utf8(&output.stdout)?
+        .split('\n')
+        .next()
+        .and_then(|line| line.split("Build Tag:        ").nth(1))
+        .ok_or(anyhow!("failed to parse version from cockroach output"))
+        .map(|version| version.into())
+}
+
+pub fn system_info(inline: bool) {
+    let mut info = vec![("kld-version", env!("CARGO_PKG_VERSION").to_string())];
```

Comment:
This is the version of kld-ctl, which is not very important. Probably the version of kld-mgr is more important than kld-ctl. We definitely need to know the version of kld.